File: solutions/2870. Minimum Number of Operations to Make Array Empty/Solution.ts
function minOperations(nums: number[]): number {
  const hash: { [s: string]: number } = {};
  let ans: number = 0;

  for (const n of nums) {
    if (!(n in hash)) hash[n] = 0;
    hash[n]++;
  }

  for (const n of Object.values(hash)) {
    if (n === 1) return -1;
    ans += Math.ceil(n / 3);
  }

  return ans;
}

File: solutions/2870. Minimum Number of Operations to Make Array Empty/Solution.py
from collections import Counter
from typing import List


class Solution:
    def minOperations(self, nums: List[int]) -> int:
        count = Counter(nums)
        ans = 0
        for c in count.values():
            if c == 1:
                return -1
            ans += (c + 2) // 3 # math.ciel(c/3)
        return ans